src/equations/compressible_euler_multicomponent_2d.jl

```diff
@@ -252,44 +357,61 @@ function initial_condition_weak_blast_wave(x, t, equations::CompressibleEulerMul
   phi               = atan(y_norm, x_norm)
   sin_phi, cos_phi  = sincos(phi)
 
-  # Calculate primitive variables.    !!! NOTE: EC flux does not transfer mass across an interface separating two different species
-  rho1              = r > 0.5 ? 0.2*1.0 : 0.2*1.1691
-  rho2              = r > 0.5 ? 0.8*1.0 : 0.8*1.1691
-  rho               = rho1 + rho2
+  # Multiple Components
+  for i in 1:length(gamma)
+    factor      = 2^(i-1) * (1-2)/(1-2^length(gamma)) # divides 1 by NCOMP, each fraction is double the previous fraction.
+    prim[i+3]   = r > 0.5 ? factor*1.0 : factor*1.1691
+  end
+
   v1                = r > 0.5 ? 0.0 : 0.1882 * cos_phi
   v2                = r > 0.5 ? 0.0 : 0.1882 * sin_phi
   p                 = r > 0.5 ? 1.0 : 1.245
 
-  return prim2cons(SVector(rho1, rho2, v1, v2, p), equations)
+  prim[1:3]         = [v1, v2, p]
+
+  result            = SVector{length(gamma)+3}(prim)
+  
+  return prim2cons(result, equations)
 end
 
 
 # Calculate 1D flux for a single point
 @inline function calcflux(u, orientation, equations::CompressibleEulerMulticomponentEquations2D)
-  @unpack cv1, cv2, gamma1, gamma2 = equations
-  rho1, rho2, rho_v1, rho_v2, rho_e = u
+  @unpack gamma, cv = equations
+  rho_v1, rho_v2, rho_e  = u
+  f         = zeros(length(gamma)+3)
+  rho       = 0
+
+  for i = 1:length(gamma)
+    rho += u[i+3]
+  end
 
-  rho   = rho1 + rho2
   v1    = rho_v1/rho
   v2    = rho_v2/rho
-  gamma = (rho1*cv1*gamma1 + rho2*cv2*gamma2) / (rho1*cv1 + rho2*cv2)
-  p     = (gamma - 1) * (rho_e - 0.5 * rho * (v1^2 + v2^2))
-  #p     = pressure(u, equations)
+  gammas= getgamma(u, equations)
+  p     = (gammas - 1) * (rho_e - 0.5 * rho * (v1^2 + v2^2))
 
   if orientation == 1
-    f1  = rho1 * v1
-    f2  = rho2 * v1
-    f3  = rho_v1 * v1 + p
-    f4  = rho_v2 * v1
-    f5  = (rho_e + p) * v1
+    for i = 1:length(gamma)
+      f[i+3] = u[i+3] * v1
+    end
+    f1  = rho_v1 * v1 + p
+    f2  = rho_v2 * v1
+    f3  = (rho_e + p) * v1
   else
-    f1  = rho1 * v2
-    f2  = rho2 * v2
-    f3  = rho_v1 * v2
-    f4  = rho_v2 * v2 + p
-    f5  = (rho_e + p) * v2
+    for i = 1:length(gamma)
+      f[i+3] = u[i+3] * v2
+    end
+    f1  = rho_v1 * v2
+    f2  = rho_v2 * v2 + p
+    f3  = (rho_e + p) * v2
   end
-  return SVector(f1, f2, f3, f4, f5)
+
+  f[1:3]  = [f1, f2, f3]
+
+  result  = SVector{length(gamma)+3}(f)
+
+  return result
 end
```

Comment:
This allocates and is likely a reason for the huge amount of allocations we observe. See `varnames` for possibilities how to fix this or ping me if you want to have some help.

Reply:
That was the culprit! Thank you. Did that for EC and ES but forgot about calcflux.